### ISSUE TITLE
GNUTLS 3.4.x

### ISSUE THREAD

**Issue by rakkesh** [Homebrew/homebrew#46562]
GnuTLS has a new release, `3.4.7`, and `3.4.x` is marked as stable. 3.4.x has API/ABI changes from the previous stable releases `3.3.x`. So, every formula which is dependent on `gnutls` needs to be revised. 
What is the process of making such an update?

By the way, I have successfully built `emacs`, `wireshark` and `qemu` with `gnutls 3.4.7`.

**Comment by rakkesh** [comment]
Formulae from `Homebrew` core which depend on GnuTLS

**Formula: Dependency type**
- [ ] bitlbee: required
- [x] emacs: optional
- [ ] freediameter: required
- [ ] git-annex: required
- [ ] glib-networking: required
- [ ] gloox: optional
- [ ] gnu-smalltalk: required
- [ ] gst-plugins-bad: optional
- [ ] gwenhywfar: required
- [ ] hamsterdb: required
- [ ] inspircd: optional
- [ ] kbtin: required
- [ ] knot: required
- [ ] libav: optional
- [ ] libinfinity: required
- [ ] libsoup: required
- [ ] libvirt: required
- [ ] libxmlsec1: recommended
- [ ] loudmouth: required
- [x] mailutils: required
- [ ] minbif: required
- [ ] ncdc: required
- [ ] net6: required
- [ ] ntopng: build
- [ ] openconnect: required
- [ ] pianobar: required
- [ ] pianod: required
- [ ] pidgin: required
- [ ] profanity: required
- [x] qemu: required
- [ ] task: required
- [ ] taskd: required
- [ ] ucommon: required
- [ ] vpnc: required
- [ ] vte3: required
- [ ] weechat: required
- [x] wireshark: required

_Note: I have built the ones with the check mark._

**Comment by DomT4** [comment]
This could get a bit complicated. To date the new GnuTLS branch has required the new Nettle [branch](https://github.com/Homebrew/homebrew-versions/blob/master/gnutls34.rb#L19) as well, so you'd have to factor in the breaking `nettle` release as well into the checks here.

**Comment by rakkesh** [comment]
Thanks @DomT4. Updated the list in my previous comment.

**Comment by rakkesh** [comment]
`lsh` failed, gist logs - https://gist.github.com/6c666e9b48ffb5d2f2c7

**Comment by rakkesh** [comment]
@DomT4 The check list is complete now. 

Bug reports:
`lsh`: reported bug[1713] on their `bugzilla`.
`net6`: https://github.com/gobby/net6/issues/1 [ project on GitHub - no activity since Nov 1, 2011 ]
`kbtin`: http://sourceforge.net/p/kbtin/bugs/11/ [ reported by @dunn but no progress ]

`gwenhywfar`: no easy way to report bugs without sending it mailing list in Deutsch.

**Comment by DomT4** [comment]
Thanks for your work here @rakkesh! Appreciate all the checking. I'd be fine with boneyarding `net6` given the lack of any commits in 4 years and new build problems, likewise with `kbtin` for the ignored bug report for 6+ months.

We should try and either backport fixes or wait on `lsh` since that's still particularly active.

**Comment by rakkesh** [comment]
Agreed, `net6` and `kbtin` can be moved to boneyard; what about `gwewnhywfar` which is used by `aqbanking`? Should these two formulae be sent to boneyard as well?

When I get time, I'll check on backporting the latest commits to `lsh` formula.

**Comment by DomT4** [comment]
> what about gwewnhywfar which is used by aqbanking? Should these two formulae be sent to boneyard as well?

Probably not. @UniqMartin may be able to help with reporting things in German, perhaps.

**Comment by UniqMartin** [comment]
> Probably not. @UniqMartin may be able to help with reporting things in German, perhaps.

Regarding `gwenhywfar` (thanks, copy & paste!): Their mailing list archive is only visible after subscription, so I haven't checked that yet, and their [bug tracker](https://devel.aqbanking.de/trac/aqbanking) looks like a really sad place: Of all the issues created there in the last 12 months, only one was closed and has seen a response from the author. So I wouldn't get my hopes up … But there's probably no harm in creating an issue and linking the build log.

**Comment by rakkesh** [comment]
[lsh](https://github.com/rakkesh/homebrew/blob/351d31aff57dc03980f639202e283446a7dbd74c/Library/Formula/lsh.rb) with patch set for `nettle3` support builds successfully with `nettle3` but not with the existing `nettle` version 2.7.1. `nettle` needs to updated before applying the patch set to `lsh` formula.

**Comment by rakkesh** [comment]
@DomT4 @UniqMartin How do you suggest we move forward on closing this issue?

**Comment by DomT4** [comment]
You can always push the `lsh` change into the same PR as the `gnutls` and `nettle` updates, and everything else that needs a revision. Make sure you annotate any patches with "Can be removed on next release" blah. Let me know before you file the PR and I'll kick the timeout in a very upwards direction.

**Comment by rakkesh** [comment]
I have done the changes, let me know a time when I can open a PR.
The commits are available here, https://github.com/rakkesh/homebrew/commits/gnutls-3.4.x.

**Comment by DomT4** [comment]
Bumped the timeout to three hours. File when desired :). Just a wee note that we don't use `args` blocks unless there are additional opt-in/opt-out arguments to be passed, Thanks!

**Comment by rakkesh** [comment]
Updated the PR as per the above comment. Removed `p11-kit` from this PR as it fails at `make check` stage [2 of 27 tests failed]. @dunn has filed a [bug report](https://bugs.freedesktop.org/show_bug.cgi?id=91602) already.

**Comment by DomT4** [comment]
Closing in favour of the open PR, Thanks!
